Modules/_io/clinic/stringio.c.h

```diff
@@ -366,6 +419,91 @@ _io_StringIO_seekable_impl(stringio *self);
 static PyObject *
 _io_StringIO_seekable(stringio *self, PyObject *Py_UNUSED(ignored))
 {
-    return _io_StringIO_seekable_impl(self);
+    PyObject *return_value = NULL;
+
+    Py_BEGIN_CRITICAL_SECTION(self);
+    return_value = _io_StringIO_seekable_impl(self);
+    Py_END_CRITICAL_SECTION();
+
+    return return_value;
+}
+
+PyDoc_STRVAR(_io_StringIO___getstate____doc__,
+"__getstate__($self, /)\n"
+"--\n"
+"\n");
+
+#define _IO_STRINGIO___GETSTATE___METHODDEF    \
+    {"__getstate__", (PyCFunction)_io_StringIO___getstate__, METH_NOARGS, _io_StringIO___getstate____doc__},
+
+static PyObject *
+_io_StringIO___getstate___impl(stringio *self);
+
+static PyObject *
+_io_StringIO___getstate__(stringio *self, PyObject *Py_UNUSED(ignored))
+{
+    PyObject *return_value = NULL;
+
+    Py_BEGIN_CRITICAL_SECTION(self);
+    return_value = _io_StringIO___getstate___impl(self);
+    Py_END_CRITICAL_SECTION();
+
+    return return_value;
+}
+
+PyDoc_STRVAR(_io_StringIO___setstate____doc__,
+"__setstate__($self, /, state)\n"
+"--\n"
+"\n");
+
+#define _IO_STRINGIO___SETSTATE___METHODDEF    \
+    {"__setstate__", _PyCFunction_CAST(_io_StringIO___setstate__), METH_FASTCALL|METH_KEYWORDS, _io_StringIO___setstate____doc__},
```

Comment:
__setstate__ should be `METH_O` not `METH_FASTCALL|METH_KEYWORDS`
See: https://github.com/python/cpython/pull/112116/files#r1398121900

Reply:
Just realized the difference, thanks a lot for point out